packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -0,0 +1,344 @@
+    }
+
+    public write(code: CodeMaker) {
+        // Before we do Anything, we need to write out our module headers, this is where
+        // we handle stuff like imports, any required initialization, etc.
+        code.line("from jsii.runtime import JSIIAssembly, JSIIMeta, jsii_method, jsii_property, jsii_classmethod, jsii_classproperty")
+
+        // Determine if we need to write out the kernel load line.
+        if (this.assembly && this.assemblyFilename) {
+            code.line(`__jsii_assembly__ = JSIIAssembly.load("${this.assembly.name}", "${this.assembly.version}", __name__, "${this.assemblyFilename}")`);
+        }
+
+        // Now that we've gotten all of the module header stuff done, we need to go
+        // through and actually write out the meat of our module.
+        for (let buffered of this.buffer) {
+            let methodName = (buffered as any)["method"] as string;
+            let args = (buffered as any)["args"] as any[];
+
+            (code as any)[methodName](...args);
+        }
+    }
+}
+
+class PythonGenerator extends Generator {
+
+    private moduleStack: Module[];
+
+    constructor(options = new GeneratorOptions()) {
+        super(options);
+
+        this.code.openBlockFormatter = s => `${s}:`;
+        this.code.closeBlockFormatter = _s => "";
+
+        this.moduleStack = [];
+    }
+
+    protected getAssemblyOutputDir(mod: spec.Assembly) {
+        return path.join("src", this.toPythonModuleName(mod.name), "_jsii");
+    }
+
+    protected onBeginAssembly(assm: spec.Assembly, _fingerprint: boolean) {
+        debug("onBeginAssembly");
+
+        // We need to write out an __init__.py for our _jsii package so that
+        // importlib.resources will be able to load our assembly from it.
+        const assemblyInitFilename = path.join(this.getAssemblyOutputDir(assm), "__init__.py");
+
+        this.code.openFile(assemblyInitFilename);
+        this.code.closeFile(assemblyInitFilename);
+    }
+
+    protected onBeginNamespace(ns: string) {
+        debug(`onBeginNamespace: ${ns}`);
+
+        const moduleName = this.toPythonModuleName(ns);
+        const loadAssembly = this.assembly.name == ns ? true : false;
+
+        let moduleArgs: any[] = [];
+
+        if (loadAssembly) {
+            moduleArgs.push([this.assembly, this.getAssemblyFileName()]);
+        }
+
+        this.moduleStack.push(new Module(moduleName, ...moduleArgs));
+    }
+
+    protected onEndNamespace(ns: string) {
+        debug(`onEndNamespace: ${ns}`);
+
+        let module = this.moduleStack.pop() as Module;
+        let moduleFilename = path.join("src", this.toPythonModuleFilename(module.name), "__init__.py");
+
+        this.code.openFile(moduleFilename);
+        module.write(this.code);
+        this.code.closeFile(moduleFilename);
+    }
+
+    protected onBeginClass(cls: spec.ClassType, abstract: boolean | undefined) {
+        debug("onBeginClass");
+
+        // TODO: Figure out what to do with abstract here.
```

Comment:
I left that as a note to myself because I wasn't sure what the semantics of an abstract class was supposed to be. Are people supposed to be able to subclass it? If so an [ABC](https://docs.python.org/3/library/abc.html) would probably be appropriate.